### PR TITLE
feat: CON-1528 Signature request context can hold the full pre-signature

### PR DESCRIPTION
**Pull request by eichhorl**
https://github.com/dfinity/ic/pull/5768 created a collection in replicated state that will be used to store available tECDSA and tSchnorr pre-signatures.

These pre-signatures are consumed by incoming signature requests. Specifically, signature request contexts are matched with available pre-signatures, by moving one pre-signature out of the pre-signature stash and including it in the context.

This PR extends the signature requests contexts for ECDSA and Schnorr to optionally hold this matched pre-signature. When the context is created initially, the pre-signature is set to `None`. The actual pairing (once a pre-signature is available) is done in a later step.

To ensure version compatibility, this field will continue to be set to `None` for now, until this change was rolled out to all subnets.

### CODE REVIEW

**Review by michael-weigelt — APPROVED** [review]

File: rs/replicated_state/src/metadata_state/tests.rs [hunk 1]
```diff
@@ -803,6 +808,100 @@ fn pre_signature_stash_roundtrip() {
...
+
+    // create some contexts with and without pre-signatures
+    for (key_id, key_transcript, pre_signature) in transcripts {
+        let arguments = match (key_id, pre_signature) {
+            (MasterPublicKeyId::Ecdsa(key_id), PreSignature::Ecdsa(ecdsa)) => {
+                let message_hash = [1; 32];
+                let args_matched = ThresholdArguments::Ecdsa(EcdsaArguments {
+                    key_id: key_id.clone(),
+                    message_hash,
+                    pre_signature: Some(EcdsaMatchedPreSignature {
```
Comment (derlerd-dfinity) [inline rs/replicated_state/src/metadata_state/tests.rs:834]: Should we also test the `None` case?

Comment (eichhorl) [reply rs/replicated_state/src/metadata_state/tests.rs:834]: Yes, note that this `match` actually returns two `ThresholdArguments`, one where the `pre_signature` is `Some(...)` (here), and one where it is `None` in line 844 below.
